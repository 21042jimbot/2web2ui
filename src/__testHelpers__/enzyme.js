export const findButton = (wrapper, buttonLabel) =>
  wrapper.findWhere(
    (node) => node.type() === 'button' && new RegExp(buttonLabel).test(node.text())
  );
