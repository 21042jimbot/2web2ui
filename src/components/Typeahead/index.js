import React from 'react';
import Downshift from 'downshift';

export default function({ items, onChange }) {

  const tafn = ({
    getInputProps,
    getItemProps,
    isOpen,
    inputValue,
    selectedItem,
    highlightedIndex
  }) => (
    <div>
    <input {...getInputProps({ placeholder: 'Favorite color ?' })} />
    {isOpen ? (
      <div style={{ border: '1px solid #ccc' }}>
        {items
          .filter(
            (i) =>
              !inputValue ||
              i.value.toLowerCase().includes(inputValue.toLowerCase()),
          )
          .map((item, index) => (
            <div
              {...getItemProps({ item, index })}
              key={item.value}
              style={{
                backgroundColor:
                  highlightedIndex === index ? 'gray' : 'white',
                fontWeight: selectedItem === item ? 'bold' : 'normal'
              }}
            >
              {item.value} [{item.type}]
            </div>
          ))}
      </div>
    ) : null}
    </div>
  );

  return (
    <Downshift onChange={onChange}>
      {tafn}
    </Downshift>
  );
}
