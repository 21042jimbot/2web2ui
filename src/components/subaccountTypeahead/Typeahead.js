import classnames from 'classnames/bind';
import Downshift from 'downshift';
import React, { Component } from 'react';
import { ActionList, Button, TextField } from '@sparkpost/matchbox';

import sortMatch from 'src/helpers/sortMatch';
import Item from './SubaccountTypeaheadItem';
import styles from './SubaccountTypeahead.module.scss';

const cx = classnames.bind(styles);

const itemToString = (item) => (item ? `${item.name} (${item.id})` : '');

export class Typeahead extends Component {
  static defaultProps = {
    name: 'subaccount'
  };

  typeaheadFn = ({
    getInputProps,
    getItemProps,
    getLabelProps,
    highlightedIndex,
    inputValue,
    openMenu,
    selectedItem,
    clearSelection,
    isOpen
  }) => {
    const { name, results, disabled, label = 'Subaccount', placeholder = (isOpen ? 'Type to search' : 'None'), error, helpText } = this.props;

    const matches = sortMatch(
      results,
      inputValue,
      (item) => `${item.name} ID: ${item.id}`
    );

    const mappedItems = (matches.length ? matches : results).map((item, index) => ({
      ...getItemProps({ item, index }),
      content: <Item name={item.name} id={item.id} />,
      highlighted: highlightedIndex === index
    }));

    const listClasses = cx('List', {
      open: isOpen && (!inputValue || matches.length),
      hasHelp: !!helpText
    });

    const textFieldProps = getInputProps({
      connectRight: selectedItem && !disabled ? this.renderClearButton(clearSelection) : null,
      readOnly: !!selectedItem,
      disabled,
      id: name,
      label,
      name,
      placeholder,
      helpText,
      error: (!isOpen && error) ? error : null
    });

    textFieldProps['data-lpignore'] = true;

    return (
      <div className={cx('Typeahead')}>
        <ActionList className={listClasses} actions={mappedItems} />
        <TextField {...textFieldProps} onFocus={openMenu} />
      </div>
    );
  };

  renderClearButton(clearSelection) {
    return <Button onClick={clearSelection}>Clear</Button>;
  }

  render() {
    const { onChange, selectedItem } = this.props;

    return (
      <Downshift
        itemToString={itemToString}
        onChange={onChange}
        selectedItem={selectedItem}
      >
        {this.typeaheadFn}
      </Downshift>
    );
  }
}

Typeahead.defaultProps = {
  results: []
};
