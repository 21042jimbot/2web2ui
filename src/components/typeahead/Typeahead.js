import classnames from 'classnames/bind';
import Downshift from 'downshift';
import React, { Component } from 'react';
import { ActionList, Button, TextField } from '@sparkpost/matchbox';

import sortMatch from 'src/helpers/sortMatch';
import styles from './Typeahead.module.scss';

const cx = classnames.bind(styles);

export const TypeaheadItem = ({ id, label }) => (
  <div className={styles.Item}>
    {label}
    <span className={styles.id}>{id}</span>
  </div>
);

export class Typeahead extends Component {
  static defaultProps = {
    name: 'subaccount'
  };

  handleStateChange = (changes, downshift) => {
    // Highlights first item in list by default
    if (!downshift.highlightedIndex) {
      downshift.setHighlightedIndex(0);
    }
  }

  typeaheadFn = ({
    getInputProps,
    getItemProps,
    getLabelProps,
    highlightedIndex,
    inputValue,
    openMenu,
    selectedItem,
    clearSelection,
    isOpen
  }) => {
    const { name, results, disabled, label, placeholder = (isOpen ? 'Type to search' : 'None'), error, helpText, errorInLabel, itemToString, renderItem } = this.props;

    const matches = sortMatch(
      results,
      inputValue,
      itemToString
    );

    const mappedItems = (matches.length ? matches : results).map((item, index) => ({
      ...getItemProps({ item, index }),
      content: renderItem ? renderItem(item) : <div className={styles.Item}>{item}</div>,
      highlighted: highlightedIndex === index
    }));

    const listClasses = cx('List', {
      open: isOpen && !selectedItem && (!inputValue || matches.length),
      hasHelp: !!helpText
    });

    const textFieldProps = getInputProps({
      connectRight: selectedItem && !disabled ? this.renderClearButton(clearSelection) : null,
      readOnly: !!selectedItem,
      disabled,
      id: name,
      label,
      name,
      placeholder,
      helpText,
      error: (!isOpen && error) ? error : null,
      errorInLabel
    });

    textFieldProps['data-lpignore'] = true;

    return (
      <div className={cx('Typeahead')}>
        <ActionList className={listClasses} actions={mappedItems} maxHeight="300" />
        <TextField {...textFieldProps} onFocus={openMenu} />
      </div>
    );
  };

  renderClearButton(clearSelection) {
    return <Button onClick={clearSelection}>Clear</Button>;
  }

  render() {
    const { itemToString, onChange, selectedItem } = this.props;

    return (
      <Downshift
        itemToString={itemToString}
        onChange={onChange}
        selectedItem={selectedItem}
        onStateChange={this.handleStateChange}
      >
        {this.typeaheadFn}
      </Downshift>
    );
  }
}

Typeahead.defaultProps = {
  results: []
};
