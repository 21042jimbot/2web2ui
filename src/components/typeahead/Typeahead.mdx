# Typeahead

Use the `<Typeahead/>` to create combobox, searchable text fields. The `<TypeaheadItem/>` can be
used to render each item in the typeahead.

## Usage

```js
import React from 'react';
import TypeAhead from 'src/components/typeAhead/TypeAhead';

const MyComponent = () => (
  <Typeahead
    name="my-typeahead"
    label="My Typeahead"
    disabled={false}
    error="This is an error"
    helpText="This is some help text"
    onChange={() => console.log('change')}
    itemToString={item => item}
    renderItem={({ label, id }) => <TypeaheadItem label={label} id={id} />}
    results={[
      {
        label: 'ABC',
        id: '123',
      },
      {
        label: 'DEF',
        id: '456',
      },
    ]}
  />
);
```
