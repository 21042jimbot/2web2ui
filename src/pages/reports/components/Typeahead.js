import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import classnames from 'classnames';
import { TextField, ActionList } from '@sparkpost/matchbox';
import TypeaheadItem from './TypeaheadItem';

import styles from './Typeahead.module.scss';

class Typeahead extends Component {
  state = {
    open: false
  }

  componentDidMount() {
    window.addEventListener('click', this.handleClickOutside);
    window.addEventListener('keydown', this.handleKey);
  }

  componentWillUnmount() {
    window.removeEventListener('click', this.handleClickOutside);
    window.removeEventListener('keydown', this.handleKey);
  }

  handleClickOutside = (e) => {
    const domNode = ReactDOM.findDOMNode(this);
    if ((!domNode || !domNode.contains(e.target))) {
      this.setState({ open: false });
    }
  }

  handleKey = (e) => {
    const code = e.code;

    if (this.state.open) {
      if (code === 'Escape') {
        this.setState({ open: false });
      }
    }
  }

  handleOnChange = (e) => {
    const { input, onChange } = this.props;

    if (e.currentTarget.value) {
      this.setState({ open: true });
    }

    if (input) {
      input.onChange(e);
    } else if (onChange) {
      onChange(e);
    }
  }

  render() {
    const {
      name = '',
      input = {},
      meta: { touched, error } = {},
      options = [],
      onSelect,
      ...rest
    } = this.props;

    const listClasses = classnames(styles.List, this.state.open && styles.open);
    const actions = options.map((option, index) => ({
      content: <TypeaheadItem value={option.value} type={option.type} />,
      onClick: () => onSelect(index)
    }));

    return (
      <div className={styles.Typeahead}>
        <div className={listClasses}>
          <ActionList actions={actions} />
        </div>
        <TextField
          id={input.name || name}
          {...rest}
          {...input}
          onChange={this.handleOnChange}
          error={touched && error ? error : undefined} />
      </div>
    );
  }
}

export default Typeahead;
