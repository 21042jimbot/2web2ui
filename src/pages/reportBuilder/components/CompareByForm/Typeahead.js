import React, { useEffect, useCallback, useState, useMemo } from 'react';
import { METRICS_API_LIMIT } from 'src/constants';
import sortMatch from 'src/helpers/sortMatch';
import { useDebouncedCallback } from 'use-debounce';

import Downshift from 'downshift';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { KeyboardArrowDown, KeyboardArrowUp } from '@sparkpost/matchbox-icons';
import { ActionList, Box, TextField } from 'src/components/matchbox';
import { LoadingSVG } from 'src/components';

const Loading = () => <LoadingSVG size="XSmall" />;

const PopoverActionList = styled(ActionList)`
  background: ${props => props.theme.colors.white};
  border: 1px solid ${props => props.theme.colors.gray['400']};
  box-shadow: ${props => props.theme.shadows['200']};
  left: 0;
  margin-top: 3px;
  max-height: 20rem;
  opacity: ${props => (props.open ? 1 : 0)};
  overflow-y: scroll;
  pointer-events: ${props => (props.open ? 'auto' : 'none')};
  visibility: ${props => (props.open ? 'visible' : 'hidden')};
  position: absolute;
  right: 0;
  top: 100%;
  transform: ${props => (props.open ? 'scale(1)' : 'scale(0.97)')};
  transition: 0.1s ease-out;
  z-index: ${props => props.theme.zIndices['overlay'] + 1};
`;

const TypeaheadWrapper = styled.div`
  margin: 0 0 1rem;
  position: relative;
  &:last-child {
    margin-bottom: 0;
  }
`;

function TypeSelect({
  disabled,
  helpText,
  id,
  itemToString,
  label,
  maxHeight,
  maxWidth,
  onChange,
  placeholder,
  renderItem,
  results,
  selectedItem,
  onInputChange,
  loading,
  suffix,
  labelHidden,
}) {
  const [matches, setMatches] = useState([]);
  // Controlled input so that we can change the value after selecting dropdown.
  const [inputValue, setInputValue] = useState(selectedItem ? itemToString(selectedItem) : '');
  const [debouncedValue, setDebouncedValue] = useState('');
  const [debounceInputChange] = useDebouncedCallback(setDebouncedValue, 300);

  useEffect(() => {
    onInputChange(debouncedValue);
  }, [onInputChange, debouncedValue]);

  useEffect(() => {
    setMatches(results);
  }, [results]);

  useEffect(() => {
    debounceInputChange(inputValue);
  }, [debounceInputChange, inputValue]);

  const handleStateChange = (changes, downshift) => {
    // Highlights first item in list by default
    if (!downshift.highlightedIndex) {
      downshift.setHighlightedIndex(0);
    }
  };

  useEffect(() => {
    if (selectedItem) {
      const selectedValue = itemToString(selectedItem);
      if (selectedValue !== inputValue) {
        setInputValue(selectedValue);
      }
    } else if (selectedItem === null) {
      setInputValue('');
    }
    // eslint-disable-next-line
  }, [selectedItem]);

  const typeaheadFn = ({
    getInputProps,
    getItemProps,
    getMenuProps,
    highlightedIndex,
    isOpen,
    selectedItem,
  }) => {
    const textFieldConfig = {
      disabled,
      label,
      labelHidden,
      placeholder,
      helpText,
      onBlur: () => {
        setInputValue(itemToString(selectedItem));
      },
      onFocus: event => {
        event.target.select();
      },
    };

    const isMenuOpen = isOpen && Boolean(debouncedValue);
    const textFieldProps = getInputProps(textFieldConfig);
    textFieldProps['data-lpignore'] = true;

    const SuffixIcon = isOpen ? KeyboardArrowUp : KeyboardArrowDown;

    return (
      <div>
        <TypeaheadWrapper>
          <Box maxWidth={maxWidth} position="relative">
            {/* hack, can't apply getMenuProps to ActionList because ref prop is not supported */}
            <div {...getMenuProps()}>
              <PopoverActionList open={isMenuOpen} maxHeight={maxHeight}>
                {loading ? (
                  <Box ml="200">
                    <Loading />
                  </Box>
                ) : (
                  <>
                    {matches.map((item, index) => (
                      <ActionList.Action
                        {...getItemProps({
                          highlighted: highlightedIndex === index,
                          index,
                          item,
                          key: item.key,
                        })}
                      >
                        {renderItem ? (
                          renderItem(item)
                        ) : (
                          <TypeaheadItem label={itemToString(item)} />
                        )}
                      </ActionList.Action>
                    ))}
                  </>
                )}
              </PopoverActionList>
            </div>
            <TextField
              suffix={suffix !== undefined ? suffix : <SuffixIcon color="blue.700" size={25} />}
              {...textFieldProps}
            />
          </Box>
        </TypeaheadWrapper>
      </div>
    );
  };

  return (
    <Downshift
      id={id}
      inputValue={inputValue}
      itemToString={itemToString}
      onChange={onChange}
      onInputValueChange={setInputValue}
      onStateChange={handleStateChange}
      selectedItem={selectedItem}
    >
      {typeaheadFn}
    </Downshift>
  );
}

TypeSelect.propTypes = {
  disabled: PropTypes.bool,
  id: PropTypes.string.isRequired,
  itemToString: PropTypes.func,
  label: PropTypes.string,
  maxHeight: PropTypes.number,
  maxNumberOfResults: PropTypes.number,
  maxWidth: PropTypes.number,
  onChange: PropTypes.func,
  placeholder: PropTypes.string,
  renderItem: PropTypes.func,
  results: PropTypes.arrayOf(
    PropTypes.shape({
      key: PropTypes.string.isRequired,
    }),
  ),
};

TypeSelect.defaultProps = {
  disabled: false,
  itemToString: item => item,
  maxHeight: 300,
  maxNumberOfResults: 100,
  maxWidth: 1200,
  placeholder: 'Type to search',
  results: [],
};

// todo, prefer to use Matchbox either Columns or Inline, but they both use <div>'s and this is rendered inside <a>'s
// see, https://css-tricks.com/flexbox-truncated-text/
const ItemContainer = styled.span`
  display: flex;
  min-width: 0;
`;

const ItemText = styled.span`
  color: ${props => props.theme.colors.gray['1000']};
  flex-grow: 1;
  flex-shrink: 1;
  font-size: ${props => props.theme.fontSizes['200']};
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
`;

const ItemMetaText = styled.span`
  color: ${props => props.theme.colors.gray['700']};
  flex-shrink: 0;
  font-size: ${props => props.theme.fontSizes['100']};
  margin-left: ${props => props.theme.space['300']};
`;

function TypeaheadItem({ label, meta }) {
  return (
    <ItemContainer>
      <ItemText>{label}</ItemText>
      {meta && <ItemMetaText>{meta}</ItemMetaText>}
    </ItemContainer>
  );
}

TypeaheadItem.propTypes = {
  label: PropTypes.string.isRequired,
  meta: PropTypes.string,
};

TypeSelect.Item = TypeaheadItem;

function Typeahead({ id, onChange, lookaheadRequest, results = [], ...rest }) {
  const [omitResults, setOmitResults] = useState(false);
  const [loading, setLoading] = useState(false);
  const [inputValue, setInputValue] = useState('');

  const updateLookahead = useCallback(
    pattern => {
      setInputValue(pattern);
      if (!lookaheadRequest) return;

      if (!pattern || pattern.length <= 2) {
        setLoading(false);
        setOmitResults(true);
        return;
      }

      setOmitResults(false);
      setLoading(true);
      const options = {
        match: pattern,
        limit: METRICS_API_LIMIT,
      };
      lookaheadRequest(options).then(() => {
        setLoading(false);
      });
    },
    [setLoading, setOmitResults, lookaheadRequest, setInputValue],
  );

  const filteredResults = useMemo(() => {
    return sortMatch(results, inputValue, a => a.value);
  }, [inputValue, results]);

  return (
    <TypeSelect
      id={id}
      onChange={onChange}
      onInputChange={updateLookahead}
      results={omitResults ? [] : filteredResults}
      itemToString={item => {
        return item ? item.value : '';
      }}
      loading={loading}
      suffix={null}
      {...rest}
    />
  );
}

export default Typeahead;
