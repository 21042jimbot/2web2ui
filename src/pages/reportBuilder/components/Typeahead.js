import React, { useCallback, useState, useMemo } from 'react';
import { ComboBoxTypeahead } from 'src/components/typeahead/AsyncComboBoxTypeahead';
import { METRICS_API_LIMIT } from 'src/constants';
import sortMatch from 'src/helpers/sortMatch';

function Typeahead(props) {
  const { dispatch, index, lookaheadRequest, reportOptions, value, results, ...rest } = props;

  const [omitResults, setOmitResults] = useState(false);
  const [loading, setLoading] = useState(false);
  const [inputValue, setInputValue] = useState('');

  const onFilterChange = useCallback(
    newValue => {
      dispatch({ type: 'ADD_FILTER', index, value: newValue });
    },
    [index, dispatch],
  );

  const updateLookahead = useCallback(
    pattern => {
      setInputValue(pattern);
      if (!lookaheadRequest) return;

      if (!pattern || pattern.length <= 2) {
        setLoading(false);
        setOmitResults(true);
        return;
      }

      setOmitResults(false);
      setLoading(true);
      const options = {
        ...reportOptions,
        match: pattern,
        limit: METRICS_API_LIMIT,
      };
      lookaheadRequest(options).then(() => {
        setLoading(false);
      });
    },
    [setLoading, setOmitResults, lookaheadRequest, reportOptions, setInputValue],
  );

  const filteredResults = useMemo(() => {
    return sortMatch(results, inputValue, a => a.value);
  }, [inputValue, results]);

  return (
    <ComboBoxTypeahead
      onChange={onFilterChange}
      onInputChange={updateLookahead}
      itemToString={item => (item ? item.value : '')}
      value={value}
      results={omitResults ? [] : filteredResults}
      loading={loading}
      {...rest}
    />
  );
}

export default Typeahead;
