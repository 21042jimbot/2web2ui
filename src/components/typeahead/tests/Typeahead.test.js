import { shallow } from 'enzyme';
import React from 'react';
import { Typeahead, TypeaheadItem } from '../Typeahead';

const results = [
  'apples',
  'bananas',
  'cauliflower'
];

describe('Typeahead', () => {
  let wrapper;

  beforeEach(() => {
    const props = {
      onChange: jest.fn(),
      itemToString: jest.fn(),
      selectedItem: jest.fn(),
      results: []
    };

    wrapper = shallow(<Typeahead {...props} />);
  });

  afterEach(() => {
    jest.resetAllMocks();
  });

  describe('render function', () => {
    let args;

    beforeEach(() => {
      wrapper.setProps({
        results
      });

      args = {
        clearSelection: jest.fn(),
        inputValue: 'test@t',
        isOpen: false,
        highlightedIndex: 0,
        getInputProps: jest.fn((props) => props),
        getItemProps: jest.fn((a) => a)
      };
    });

    it('should render the list', () => {
      const result = shallow(wrapper.instance().typeaheadFn(args));
      expect(result).toMatchSnapshot();
    });

    it('should switch placeholder if open', () => {
      const result = shallow(wrapper.instance().typeaheadFn({ ...args, isOpen: true }));
      expect(result.find('TextField').props().placeholder).toEqual('Type to search');
    });

    it('should render with selected item and not disabled', () => {
      const result = shallow(wrapper.instance().typeaheadFn({ ...args, selectedItem: results[2] }));
      expect(result.find('TextField').props().connectRight).toMatchSnapshot(); // Clear button
      expect(result.find('TextField').props().readOnly).toBe(true); // Boolean prop
    });

    it('should not render clear button if disabled with selected item', () => {
      wrapper.setProps({ disabled: true });
      const result = shallow(wrapper.instance().typeaheadFn({ ...args, selectedItem: results[2] }));
      expect(result.find('TextField').props().connectRight).toBe(null); // No clear button
    });

    it('should render error', () => {
      wrapper.setProps({ error: 'an error omg' });
      const result = shallow(wrapper.instance().typeaheadFn(args));
      expect(result.find('TextField').props().error).toBe('an error omg');
    });

    it('should not render error if it is open', () => {
      wrapper.setProps({ error: 'an error omg' });
      const result = shallow(wrapper.instance().typeaheadFn({ ...args, isOpen: true }));
      expect(result.find('TextField').props().error).toBe(null);
    });

    it('should select first result when typeahead opens', () => {
      const ds = {
        setHighlightedIndex: jest.fn(),
        highlightedIndex: null
      };
      wrapper.instance().handleStateChange(null, ds);
      expect(ds.setHighlightedIndex).toHaveBeenCalledWith(0);
    });

    it('renders a typeahead item', () => {
      const wrapper = shallow(<TypeaheadItem id="example-id" label="Example Label" />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
