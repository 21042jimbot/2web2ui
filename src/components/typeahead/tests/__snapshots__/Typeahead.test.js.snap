// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typeahead render function should render the list 1`] = `
<div
  className="Typeahead"
>
  <ActionList
    actions={
      Array [
        Object {
          "content": <div
            className={undefined}
          >
            apples
          </div>,
          "highlighted": true,
          "index": 0,
          "item": "apples",
        },
        Object {
          "content": <div
            className={undefined}
          >
            bananas
          </div>,
          "highlighted": false,
          "index": 1,
          "item": "bananas",
        },
        Object {
          "content": <div
            className={undefined}
          >
            cauliflower
          </div>,
          "highlighted": false,
          "index": 2,
          "item": "cauliflower",
        },
      ]
    }
    className="List"
  />
  <TextField
    connectRight={null}
    data-lpignore={true}
    error={null}
    placeholder="None"
    readOnly={false}
    required={false}
    resize="both"
    type="text"
  />
</div>
`;

exports[`Typeahead render function should render with selected item and not disabled 1`] = `
<Button
  onClick={[MockFunction]}
  size="default"
>
  Clear
</Button>
`;
